src/governance/governor/Governor.sol

```diff
@@ -75,45 +75,36 @@ contract Governor is IGovernor, VersionedContract, UUPS, Ownable, EIP712, Propos
     /// @notice Initializes a DAO's governor
     /// @param _treasury The DAO's treasury address
     /// @param _token The DAO's governance token address
-    /// @param _vetoer The address eligible to veto proposals
-    /// @param _votingDelay The voting delay
-    /// @param _votingPeriod The voting period
-    /// @param _proposalThresholdBps The proposal threshold basis points
-    /// @param _quorumThresholdBps The quorum threshold basis points
-    function initialize(
-        address _treasury,
-        address _token,
-        address _vetoer,
-        uint256 _votingDelay,
-        uint256 _votingPeriod,
-        uint256 _proposalThresholdBps,
-        uint256 _quorumThresholdBps
-    ) external initializer {
+    /// @param _data The encoded governor parameters
+    function initialize(address _treasury, address _token, bytes calldata _data) external initializer {
```

Comment:
could pass in the generic struct here and _data as a part of that to prevent re-encoding and passing the calldata so many times.

Reply:
To create a struct with the addresses and data parameters we would need to copy the _data param (from calldata) and deployed addresses (from storage) to memory which might be more expensive than re-encodinga and passing the calldata I think